impeller/entity/contents/runtime_effect_contents.cc

```diff
@@ -0,0 +1,164 @@
+                                    ShaderStage::kFragment);
+    if (!function) {
+      VALIDATION_LOG
+          << "Failed to fetch runtime effect function immediately after "
+             "registering it (entry point: "
+          << runtime_stage_->GetEntrypoint() << ")";
+      return false;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  /// Resolve geometry.
+  ///
+
+  auto geometry_result = GetGeometry()->GetPositionBuffer(
+      context->GetResourceAllocator(), pass.GetTransientsBuffer(),
+      renderer.GetTessellator(), pass.GetRenderTargetSize());
+
+  //--------------------------------------------------------------------------
+  /// Get or create runtime stage pipeline.
+  ///
+
+  using VS = PositionNoColorVertexShader;
+  PipelineDescriptor desc;
+  desc.SetLabel("Runtime Stage");
+  desc.AddStageEntrypoint(
+      library->GetFunction(VS::kEntrypointName, ShaderStage::kVertex));
+  desc.AddStageEntrypoint(library->GetFunction(runtime_stage_->GetEntrypoint(),
+                                               ShaderStage::kFragment));
+  auto vertex_descriptor = std::make_shared<VertexDescriptor>();
+  if (!vertex_descriptor->SetStageInputs(VS::kAllShaderStageInputs)) {
+    VALIDATION_LOG << "Failed to set stage inputs for runtime effect pipeline.";
+  }
+  desc.SetVertexDescriptor(std::move(vertex_descriptor));
+  desc.SetColorAttachmentDescriptor(0u, {.format = PixelFormat::kDefaultColor});
+  desc.SetStencilAttachmentDescriptors({});
+  desc.SetStencilPixelFormat(PixelFormat::kDefaultStencil);
+
+  auto options = OptionsFromPassAndEntity(pass, entity);
+  if (geometry_result.prevent_overdraw) {
+    options.stencil_compare = CompareFunction::kEqual;
+    options.stencil_operation = StencilOperation::kIncrementClamp;
+  }
+  options.ApplyToPipelineDescriptor(desc);
+
+  auto pipeline = context->GetPipelineLibrary()->GetPipeline(desc).get();
+  if (!pipeline) {
+    VALIDATION_LOG << "Failed to get or create runtime effect pipeline.";
+    return false;
+  }
+
+  Command cmd;
+  cmd.label = "RuntimeEffectContents";
+  cmd.pipeline = pipeline;
+  cmd.stencil_reference = entity.GetStencilDepth();
+  cmd.BindVertices(geometry_result.vertex_buffer);
+  cmd.primitive_type = geometry_result.type;
+
+  //--------------------------------------------------------------------------
+  /// Vertex stage uniforms.
+  ///
+
+  VS::VertInfo frame_info;
+  frame_info.mvp = Matrix::MakeOrthographic(pass.GetRenderTargetSize()) *
+                   entity.GetTransformation();
+  VS::BindVertInfo(cmd, pass.GetTransientsBuffer().EmplaceUniform(frame_info));
+
+  //--------------------------------------------------------------------------
+  /// Fragment stage uniforms.
+  ///
+
+  size_t buffer_index = 0;
+  for (auto uniform : runtime_stage_->GetUniforms()) {
+    // TODO(bdero): Populate this metadata once GLES is able to handle
```

Comment:
Please file (or update) an issue for this TODO

Reply:
Done: https://github.com/flutter/flutter/issues/113715